benchmarks/set_http_meta/scenario.py

```diff
@@ -0,0 +1,132 @@
+    "HTTP_COOKIE": "Pycharm-45729245=449f1b16-fe0a-4623-92bc-418ec418ed4b; Idea-9fdb9ed8="
+    "448d4c93-863c-4e9b-a8e7-bbfbacd073d2; csrftoken=cR8TVoVebF2afssCR16pQeqHcxA"
+    "lA3867P6zkkUBYDL5Q92kjSGtqptAry1htdlL; _xsrf=2|d4b85683|7e2604058ea673d12dc6604f"
+    '96e6e06d|1635869800; username-localhost-8888="2|1:0|10:1637328584|23:username-loca'
+    "lhost-8888|44:OWNiOTFhMjg1NDllNDQxY2I2Y2M2ODViMzRjMTg3NGU=|3bc68f938dcc081a9a02e51660"
+    '0c0d38b14a3032053a7e16b180839298e25b42"',
+    "wsgi.input": BytesIO(),
+    "wsgi.url_scheme": "http",
+}
+
+COOKIES = {"csrftoken": "cR8TVoVebF2afssCR16pQeqHcxAlA3867P6zkkUBYDL5Q92kjSGtqptAry1htdlL"}
+
+DATA_GET = dict(
+    method="GET",
+    url="http://localhost:8888{}".format(PATH),
+    status_code=200,
+    status_msg="OK",
+    query=b"key1=value1&key2=value2&token=cR8TVoVebF2afssCR16pQeqHcxAlA3867P6zkkUBYDL5Q92kjSGtqptAry1htdlL",
+    parsed_query={
+        "key1": "value1",
+        "key2": "value2",
+        "token": "cR8TVoVebF2afssCR16pQeqHcxAlA3867P6zkkUBYDL5Q92kjSGtqptAry1htdlL",
+    },
+    request_headers=COMMON_DJANGO_META,
+    response_headers=COMMON_DJANGO_META,
+    retries_remain=0,
+    raw_uri="http://localhost:8888{}?key1=value1&key2=value2&token="
+    "cR8TVoVebF2afssCR16pQeqHcxAlA3867P6zkkUBYDL5Q92kjSGtqptAry1htdlL".format(PATH),
+    request_cookies=COOKIES,
+    request_path_params={"id": 1},
+)
+
+DATA_POST = dict(
+    method="GET",
+    url="http://localhost:8888/test-benchmark",
+    status_code=200,
+    status_msg="OK",
+    query=b"key1=value1&key2=value2",
+    parsed_query={"key1": "value1", "key2": "value2"},
+    request_headers=COMMON_DJANGO_META,
+    response_headers=COMMON_DJANGO_META,
+    retries_remain=0,
+    raw_uri="http://127.0.0.1:8081/test/1/?key1=value1&key2=value2",
+    request_cookies=COOKIES,
+    request_path_params={"id": 1},
+    request_body=None,
+)
+
+
+class SetHttpMeta(bm.Scenario):
+    allenabled = bm.var_bool()
+    useragentkey = bm.var_bool()
+    useragentvariants = bm.var_bool()
+
+    def run(self):
+        # run scenario to also set tags on spans
+        if self.allenabled:
+            config = Config(lambda: True)
+        else:
+            config = Config(lambda: False)
+
+        data = copy.deepcopy(DATA_GET)
+
+        user_agent_keys = []
+        if self.useragentkey:
+            user_agent_keys = ["HTTP_USER_AGENT"]
+            if self.useragentvariants:
+                user_agent_keys = ["HTTP_USER_AGENT", "USER_AGENT", "UserAgent", "User-Agent", "user-agent"]
+
+        def bm(loops):
+            for _ in range(loops):
+                for user_agent_key in user_agent_keys:
+                    data["request_headers"][user_agent_key] = "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 " \
+                                                              "(KHTML, like Gecko) Chrome/92.0.4515.159 Safari/537.36"
+                    span = utils.gen_span(str("test"))
```

Comment:
is the `str()` needed?

Reply:
it'll be better if we can create the span once outside of this loop to avoid measuring the cost of creating the span (some isinstance checks, and random number generator, and getting current timestamp.

Reply:
We can follow a similar approach taken in other scenarios if we are trying to isolate regressions in a specific function call, as is the case here.

https://github.com/DataDog/dd-trace-py/blob/d32de4e02e8038a49c9c8aa9e33cc6335408eeb4/benchmarks/http_propagation_inject/scenario.py#L24-L30